indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1072,9 +1105,26 @@ private void addSequence(final SequenceMetadata<PartitionIdType, SequenceOffsetT
       }
     }
 
+    if (!isEndOffsetExclusive() && !sequences.isEmpty()) {
```

Comment:
Should this be moved into the check above since both blocks have `!sequences.isEmpty()`?

Reply:
I thought it too, but I decided to check separately because `sequences` is a `CopyOnWriteArrayList` and so it can be updated between lines.